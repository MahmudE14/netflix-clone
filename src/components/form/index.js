import { Container, Base, Error, Title, Text, TextSmall, Link, Input, Submit } from './styles/form'

export default function Form({ children, ...restProps }) {
    return (
        <Container {...restProps}>{children}</Container>
    )
};

Form.Base = function FormContainer({ children, ...restProps }) {
    return <Base { ...restProps }>{children}</Base>
}

Form.Error = function FormContainer({ children, ...restProps }) {
    return <Error { ...restProps }>{children}</Error>
}

Form.Title = function FormContainer({ children, ...restProps }) {
    return <Title { ...restProps }>{children}</Title>
}

Form.Text = function FormContainer({ children, ...restProps }) {
    return <Text { ...restProps }>{children}</Text>
}

Form.TextSmall = function FormContainer({ children, ...restProps }) {
    return <TextSmall { ...restProps }>{children}</TextSmall>
}

Form.Link = function FormContainer({ children, ...restProps }) {
    return <Link { ...restProps }>{children}</Link>
}

Form.Input = function FormContainer({ children, ...restProps }) {
    return <Input { ...restProps }>{children}</Input>
}

Form.Submit = function FormContainer({ children, ...restProps }) {
    return <Submit { ...restProps }>{children}</Submit>
}
