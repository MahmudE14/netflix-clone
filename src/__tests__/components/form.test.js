import { render, screen } from "@testing-library/react";
import { Form } from "../../components";
import { Link as ReactRouterLink } from "react-router-dom";

jest.mock('react-router-dom')

describe("<Form />", () => {
  it("render the <Form /> with populated data", () => {
    const { container } = render(
      <Form>
        <Form.Title>Sign In Now</Form.Title>
        <Form.Base>
          <Form.Input placeholder="Email address" onChange={() => {}} />
          <Form.Input
            type="password"
            autoComplete="off"
            placeholder="Password"
            onChange={() => {}}
          />
          <Form.Submit disabled data-testid="sign-in">
            Sign In
          </Form.Submit>
        </Form.Base>

        <Form.Text>
          New to Netflix? <Form.Link to="/signup">Sign up now.</Form.Link>
        </Form.Text>
        <Form.TextSmall>
          This page is protected by Google reCAPTCHA to ensure you're not a bot.
          Learn more.
        </Form.TextSmall>
      </Form>
    );

    expect(screen.getByText("This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.")).toBeTruthy()
    expect(screen.getByText("Sign In Now")).toBeTruthy()
    expect(screen.getByText("Sign In").disabled).toBeTruthy()
    expect(screen.getByPlaceholderText("Email address")).toBeTruthy()
    expect(screen.getByPlaceholderText("Password")).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  });

  it("render the <Form /> with and error", () => {
    const { container } = render(
        <Form>
            <Form.Error>You email addresss is already being used</Form.Error>
            <Form.Submit type="submit">Sign In</Form.Submit>
        </Form>
    )

    expect(screen.getByText('You email addresss is already being used')).toBeTruthy()
    expect(screen.queryByText('Sign In').disabled).toBeFalsy()

    expect(container.firstChild).toMatchSnapshot()
  })
});
