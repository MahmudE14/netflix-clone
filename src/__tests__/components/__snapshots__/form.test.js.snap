// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> render the <Form /> with and error 1`] = `
<div
  class="form__Container-sc-1jfxee4-0 liHZUV"
>
  <div
    class="form__Error-sc-1jfxee4-1 lduXan"
  >
    You email addresss is already being used
  </div>
  <button
    class="form__Submit-sc-1jfxee4-8 gtFHEj"
    type="submit"
  >
    Sign In
  </button>
</div>
`;

exports[`<Form /> render the <Form /> with populated data 1`] = `
<div
  class="form__Container-sc-1jfxee4-0 liHZUV"
>
  <h1
    class="form__Title-sc-1jfxee4-3 djozki"
  >
    Sign In Now
  </h1>
  <form
    class="form__Base-sc-1jfxee4-2 kaAhuR"
  >
    <input
      class="form__Input-sc-1jfxee4-7 ggOTfY"
      placeholder="Email address"
    />
    <input
      autocomplete="off"
      class="form__Input-sc-1jfxee4-7 ggOTfY"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-1jfxee4-8 gtFHEj"
      data-testid="sign-in"
      disabled=""
    >
      Sign In
    </button>
  </form>
  <p
    class="form__Text-sc-1jfxee4-4 fRUcIj"
  >
    New to Netflix? 
  </p>
  <p
    class="form__TextSmall-sc-1jfxee4-5 hVIgtL"
  >
    This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
  </p>
</div>
`;
